packages/ubuntu_desktop_installer/lib/pages/installation_type/installation_type_page.dart

```diff
@@ -51,7 +51,9 @@ class _InstallationTypePageState extends State<InstallationTypePage> {
       case 1:
         return lang.installationTypeOSDetected(os.single.long);
       case 2:
+        if (os.hasDuplicates) continue duplicates;
```

Comment:
Honest question: do you think this jump reads better than leaving it fall through the default case?

```dart
  if (!os.hasDuplicates) {
    return lang.installationTypeDualOSDetected(os.first.long, os.last.long);
  }

default:
    return lang.installationTypeMultiOSDetected;
```

Reply:
That was actually my first attempt too but the Dart analyzer doesn't allow falling through unless the case is empty ([switch_case_completes_normally](https://dart.dev/tools/diagnostic-messages?utm_source=dartdev&utm_medium=redir&utm_id=diagcode&utm_content=switch_case_completes_normally#switch_case_completes_normally)).

Do you like something like this more?
```diff
diff --git a/packages/ubuntu_desktop_installer/lib/pages/installation_type/installation_type_page.dart b/packages/ubuntu_desktop_installer/lib/pages/installation_type/installation_type_pag>
index 85a143f9..01c76bb0 100644
--- a/packages/ubuntu_desktop_installer/lib/pages/installation_type/installation_type_page.dart
+++ b/packages/ubuntu_desktop_installer/lib/pages/installation_type/installation_type_page.dart
@@ -51,9 +51,9 @@ class _InstallationTypePageState extends State<InstallationTypePage> {
       case 1:
         return lang.installationTypeOSDetected(os.single.long);
       case 2:
-        if (os.hasDuplicates) continue duplicates;
-        return lang.installationTypeDualOSDetected(os.first.long, os.last.long);
-      duplicates:
+        return os.hasDuplicates
+            ? lang.installationTypeMultiOSDetected
+            : lang.installationTypeDualOSDetected(os.first.long, os.last.long);
       default:
         return lang.installationTypeMultiOSDetected;
     }
@@ -68,10 +68,10 @@ class _InstallationTypePageState extends State<InstallationTypePage> {
       case 1:
         return lang.installationTypeAlongside(product, os.single.long);
       case 2:
-        if (os.hasDuplicates) continue duplicates;
-        return lang.installationTypeAlongsideDual(
-            product, os.first.long, os.last.long);
-      duplicates:
+        return os.hasDuplicates
+            ? lang.installationTypeAlongsideMulti(product)
+            : lang.installationTypeAlongsideDual(
+                product, os.first.long, os.last.long);
       default:
         return lang.installationTypeAlongsideMulti(product);
     }
```

Reply:
Honestly I do.

Reply:
how the heck can we make this pretty :D
```dart
return os.length > 2 || os.hasDuplicates
    ? lang.installationTypeMultiOSDetected
    : os.length == 2
        ? lang.installationTypeDualOSDetected(os.first.long, os.last.long)
        : os.length == 1
            ? lang.installationTypeOSDetected(os.single.long)
            : lang.installationTypeNoOSDetected;
```

Reply:
meh...
```dart
final count = os.hasDuplicates ? -1 : os.length;
switch (count) {
  case 0:
    return lang.installationTypeNoOSDetected;
  case 1:
    return lang.installationTypeOSDetected(os.single.long);
  case 2:
    return lang.installationTypeDualOSDetected(os.first.long, os.last.long);
  default:
    return lang.installationTypeMultiOSDetected;
}
```

Reply:
That would work as well. I like the ternary as is, though.

Reply:
Thanks, it looks more readable than the first proposal :)